api/app/registration/api.py

```diff
@@ -1,21 +1,191 @@
 from datetime import datetime
-import traceback
-from flask_restful import fields, marshal_with
 import flask_restful as restful
 from sqlalchemy.exc import IntegrityError
-from sqlalchemy.exc import SQLAlchemyError
-from app.utils import errors
-from app import LOGGER
 from app.registration.models import RegistrationSection
 from app.registration.models import RegistrationQuestion
 from app.registration.models import RegistrationForm
-from app.registration.models import Offer
 from app.registration.mixins import RegistrationFormMixin, RegistrationSectionMixin, RegistrationQuestionMixin
+from app.utils.auth import verify_token
+import traceback
+from flask import g, request
+from flask_restful import  fields, marshal_with
+from sqlalchemy.exc import SQLAlchemyError
 from app.events.models import Event
+from app.registration.models import Offer
+from app.registration.mixins import OfferMixin
 from app.users.models import AppUser
-from app import db
-from app.utils.auth import auth_required, admin_required, verify_token
-from flask import g, request
+from app import db, LOGGER
+from app.utils import errors
+from app.utils.auth import auth_required, admin_required
+from app.utils.emailer import send_mail
+
+OFFER_EMAIL_BODY = """
+Dear {} {} {},
+
+Congratulations on you offer
+
+Offer Details \n
+user_id {} \n
+event_id {} \n
+offer_date {} \n
+expiry_date {} \n
+payment_required {} \n
+travel_award {} \n
+accommodation_award {} \n
+
+Kind Regards,
+The Baobab Team
+"""
+
+
+def offer_info(offer_entity):
+    return {
+        'user_id': offer_entity.user_id,
+        'event_id': offer_entity.event_id,
+        'offer_date': offer_entity.offer_date,
+        'expiry_date': offer_entity.expiry_date,
+        'payment_required': offer_entity.payment_required,
+        'travel_award': offer_entity.travel_award,
+        'accommodation_award': offer_entity.accommodation_award
+    }
+
+
+def offer_update_info(offer_entity):
+    return {
+        'user_id': offer_entity.user_id,
+        'event_id': offer_entity.event_id,
+        'offer_date': offer_entity.offer_date,
+        'expiry_date': offer_entity.expiry_date,
+        'payment_required': offer_entity.payment_required,
+        'travel_award': offer_entity.travel_award,
+        'accommodation_award': offer_entity.accommodation_award,
+        'rejected': offer_entity.rejected,
+        'accepted': offer_entity.accepted,
+        'rejected_reason': offer_entity.rejected_reason,
+        'updated_at': offer_entity.updated_at,
+    }
+
+
+class OfferAPI(OfferMixin, restful.Resource):
+    offer_fields = {
+        'user_id': fields.Integer,
+        'event_id': fields.Integer,
+        'offer_date': fields.DateTime,
```

Comment:
You can use fields.DateTime('iso8601') to set the (de)serialization format - and for consistency with other controllers